### PR TITLE
feat: support etcd username&&password

### PR DESCRIPTION
**Pull request by qiqizjl**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance

- Related issues

___
### Bugfix
- Description
support ETCD username password
- How to fix?

___
### New feature or improvement
- Describe the details and related test reports.

### CODE REVIEW

**Comment by nic-chen** [comment]
Good job.

Need test case for it. Thanks.

**Comment by nic-chen** [comment]
I think we could add test cases in:
https://github.com/apache/apisix-dashboard/blob/v2.0/api/test/shell/cli_test.sh

and verify auth config like:
https://github.com/apache/apisix-dashboard/blob/v2.0/api/test/shell/docker_deploy_test.sh#L30-L43

File: api/conf/conf.yaml [hunk 1]
```diff
@@ -22,6 +22,8 @@ conf:
   etcd:
     endpoints: # supports defining multiple etcd host addresses for an etcd cluster
       - 127.0.0.1:2379
+    username: ""
```
Comment (juzhiyuan) [inline api/conf/conf.yaml:25]: if we use an empty string here, does it mean we could connect ETCD anonymously?

File: api/internal/core/storage/etcd.go [hunk 2]
```diff
@@ -32,11 +32,16 @@ var (
 type EtcdV3Storage struct {
 }
 
-func InitETCDClient(endpoints []string) error {
-	cli, err := clientv3.New(clientv3.Config{
+func InitETCDClient(endpoints []string, username, password string) error {
```
Comment (juzhiyuan) [inline api/internal/core/storage/etcd.go:35]: if we set typing for username?

File: api/conf/conf.yaml (hunk 1 above)
Comment (membphis) [reply api/conf/conf.yaml:25]: that is not a good way

File: api/internal/core/storage/etcd.go (hunk 2 above)
Comment (qiqizjl) [reply api/internal/core/storage/etcd.go:35]: username string,password string = username,password string

File: api/conf/conf.yaml (hunk 1 above)
Comment (qiqizjl) [reply api/conf/conf.yaml:25]: Yes. When the string is empty, it can be linked anonymously

**Comment by membphis** [comment]
@qiqizjl please add test case

**Comment by nic-chen** [comment]
gentle ping @qiqizjl

**Comment by membphis** [comment]
@qiqizjl any news?

**Comment by qiqizjl** [comment]
Trying to add test case

**Comment by juzhiyuan** [comment]
Will this close https://github.com/apache/apisix-dashboard/issues/839?

**Comment by membphis** [comment]
@qiqizjl any news? ^_^

**Comment by nic-chen** [comment]
we have support this feature by other PR #951 . closing this now.

**Pull request closed without merging**